packages/react-reconciler/src/ReactFiberBeginWork.new.js

```diff
@@ -3480,14 +3482,23 @@ function beginWork(
       // If props or context changed, mark the fiber as having performed work.
       // This may be unset if the props are determined to be equal later (memo).
       didReceiveUpdate = true;
-    } else if (!includesSomeLane(renderLanes, updateLanes)) {
-      didReceiveUpdate = false;
-      return attemptEarlyBailoutIfNoScheduledUpdate(
-        current,
-        workInProgress,
-        renderLanes,
-      );
     } else {
+      // Neither props nor legacy context changes. Check if there's a pending
+      // update or context change.
+      if (
+        !checkScheduledUpdateOrContext(current, renderLanes) &&
```

Comment:
Nit: this would be easier for me to read if it started with `has`

"not check" is harder to skim than "not has"

Reply:
Yeah I suppose, I did it like that for consistency with the existing `checkIfWorkInProgressReceivedUpdate` function, which I think was originally named like that to avoid a clash with the variable it returned... which was named `hasWorkInProgressReceivedUpdate`

Reply:
How about I'll assign to a named variable before negating it :D

Reply:
sgtm haha